components/autofill/src/error.rs

```diff
@@ -44,4 +70,82 @@ pub enum Error {
     NoSuchRecord(String),
 }
 
-pub type Result<T> = std::result::Result<T, Error>;
+// Define how our internal errors are handled and converted to external errors.
+impl GetErrorHandling for Error {
+    type ExternalError = AutofillApiError;
+
+    // Return how to handle our internal errors
+    fn get_error_handling(&self) -> ErrorHandling<Self::ExternalError> {
+        // WARNING: The details inside the `AutofillApiError` we return should not
+        // contain any personally identifying information.
+        // However, because many of the string details come from the underlying
+        // internal error, we operate on a best-effort basis, since we can't be
+        // completely sure that our dependencies don't leak PII in their error
+        // strings.  For example, `rusqlite::Error` could include data from a
+        // user's database in their errors, which would then cause it to appear
+        // in our `AutofillApiError::SqlError` structs, log messages, etc.
+        // But because we've never seen that in practice we are comfortable
+        // forwarding that error message into ours without attempting to sanitize.
```

Comment:
I copied the PII note from logins to each component since it seems useful and I'm afraid if I put it in some centralized place like the error-support README it would get overlooked.  But maybe it's overkill, WDTY?

Reply:
I'm torn on this too :) Also torn whether it's better here or near the actual public enum?

An alternative might be to move some of the text into the new error doc you wrote and in each component make the note much shorter and link to that doc? I'm fine either way though.

Reply:
This makes a lot of sense to me, since there's also other information that could be useful there.